subsys/bluetooth/host/gatt.c

```diff
@@ -4684,6 +4698,13 @@ int bt_gatt_subscribe(struct bt_conn *conn,
 		}
 	}
 
+	/* if application didn't set required minimum seucrity for notification
+	 * use sec level from time of subscription
+	 */
+	if (params->min_security < BT_SECURITY_L1) {
+		params->min_security = conn->sec_level;
+	}
+
```

Comment:
there is also BT_SECURITY_L0 == 0

Reply:
But is it an issue if the application sets `min_security` to 0? I think it would do that in most cases.

Reply:
Can you elaborate a bit more? I fail to see the issue of the application setting this to L0 (0) without the stack overwriting it. 

If anything if we do overwrite it, then it should be `params->min_security = BT_SECURITY_L1`, otherwise if the intended security was a low as possible, the the `conn->sec_level` is > L1, then we unnecessarily  set it too high.